src/subclass/cell_renderer.rs

```diff
@@ -0,0 +1,658 @@
+use glib::ObjectClass;
+use glib::GString;
+
+use CellEditable;
+use CellRenderer;
+use CellRendererClass;
+use CellRendererState;
+use SizeRequestMode;
+use Widget;
+
+pub trait CellRendererImpl: CellRendererImplExt + ObjectImpl + 'static {
+    fn get_request_mode(&self, renderer: &CellRenderer) -> SizeRequestMode {
+        self.parent_get_request_mode(renderer)
+    }
+
+    fn get_preferred_width<P: IsA<Widget>>(
+        &self,
+        renderer: &CellRenderer,
+        widget: &P,
+    ) -> (i32, i32) {
+        self.parent_get_preferred_width(renderer, widget)
+    }
+
+    fn get_preferred_width_for_height<P: IsA<Widget>>(
+        &self,
+        renderer: &CellRenderer,
+        widget: &P,
+        height: i32,
+    ) -> (i32, i32) {
+        self.parent_get_preferred_width_for_height(renderer, widget, height)
+    }
+
+    fn get_preferred_height<P: IsA<Widget>>(
+        &self,
+        renderer: &CellRenderer,
+        widget: &P,
+    ) -> (i32, i32) {
+        self.parent_get_preferred_height(renderer, widget)
+    }
+
+    fn get_preferred_height_for_width<P: IsA<Widget>>(
+        &self,
+        renderer: &CellRenderer,
+        widget: &P,
+        width: i32,
+    ) -> (i32, i32) {
+        self.parent_get_preferred_height_for_width(renderer, widget, width)
+    }
+
+    fn get_aligned_area<P: IsA<Widget>>(
+        &self,
+        renderer: &CellRenderer,
+        widget: &P,
+        flags: CellRendererState,
+        cell_area: &gdk::Rectangle,
+    ) -> gdk::Rectangle {
+        self.parent_get_aligned_area(renderer, widget, flags, cell_area)
+    }
+
+    fn render<P: IsA<Widget>>(
+        &self,
+        renderer: &CellRenderer,
+        cr: &cairo::Context,
+        widget: &P,
+        background_area: &gdk::Rectangle,
+        cell_area: &gdk::Rectangle,
+        flags: CellRendererState,
+    ) {
+        self.parent_render(renderer, cr, widget, background_area, cell_area, flags);
+    }
+
+    fn activate<P: IsA<Widget>>(
+        &self,
+        renderer: &CellRenderer,
+        event: &gdk::Event,
+        widget: &P,
+        path: &str,
+        background_area: &gdk::Rectangle,
+        cell_area: &gdk::Rectangle,
+        flags: CellRendererState,
+    ) -> bool {
+        self.parent_activate(
+            renderer,
+            event,
+            widget,
+            path,
+            background_area,
+            cell_area,
+            flags,
+        )
+    }
+
+    fn start_editing<P: IsA<Widget>>(
+        &self,
+        renderer: &CellRenderer,
+        event: Option<&gdk::Event>,
+        widget: &P,
+        path: &str,
+        background_area: &gdk::Rectangle,
+        cell_area: &gdk::Rectangle,
+        flags: CellRendererState,
+    ) -> Option<CellEditable> {
+        self.parent_start_editing(
+            renderer,
+            event,
+            widget,
+            path,
+            background_area,
+            cell_area,
+            flags,
+        )
+    }
```

Comment:
There's also `editing_chancelled` and `editing_started`

Reply:
ah right. I forgot why I left out the signal, but it was part of me not knowing what I was doing.

Reply:
No worries :) These are virtual functions for the default handlers of these signals. With regard to subclassing you handle them exactly like others, doesn't matter that the signal machinery calls into them instead of other code.

Reply:
This is done now.